plugins/styleguide/assets/javascripts/discourse/components/styleguide-icons.js.es6

```diff
@@ -13,8 +14,23 @@ export default Ember.Component.extend({
   @afterRender
   setIconIds() {
     let symbols = document.querySelectorAll("#svg-sprites symbol");
-    let ids = Array.from(symbols).mapBy("id");
-
-    this.set("iconIds", ids);
+    if (symbols.length > 0) {
+      let ids = Array.from(symbols).mapBy("id");
+      this.set("iconIds", ids.sort());
+    } else {
+      // Let's try again a short time later if there are no svgs loaded yet
+      later(
+        this,
+        () => {
+          this.set(
+            "iconIds",
+            Array.from(document.querySelectorAll("#svg-sprites symbol"))
+              .mapBy("id")
+              .sort()
+          );
+        },
+        1500
+      );
```

Comment:
Can this be simplified like this or is there a risk of looping infinitely?

```suggestion
      later(this, this.setIconIds, 1500);
```

Reply:
That seems to work quite well.